python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -911,9 +949,9 @@ def _create_pattern_spokes(self, src_ctr, src_delta_array, src_dist_array,
 
         return output_spokes
 
-    def _test_spoke(self, src_ctr, src_delta, src_dist, src_ctr_delta,
-                    src_ctr_dist, ref_ctr, ref_ctr_id, ref_delta, ref_dist,
-                    ref_dist_idx_array, ref_delta_array, ref_dist_array,
+    def _test_spoke(self, cos_theta_src, sin_theta_src, ref_ctr, ref_ctr_id,
+                    proj_ref_ctr_delta, proj_ref_ctr_dist_sq,
+                    ref_dist_idx_array, ref_delta_array,
```

Comment:
It seems a bit odd to pass in cos and sin theta instead of computing them within the function, as you could mistakenly pass in inconsistent values (i.e. both zero). Why not pass the two vectors instead? If you're doing it to avoid repeated evaluation of cos/sin theta then perhaps the function should allow for iterable inputs.